clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -559,8 +545,6 @@ private long endTimestamp() {
         } catch (KafkaException e) {
             if (fetched.isEmpty())
                 throw e;
-            // To be thrown in the next call of this method
-            nextInLineExceptionMetadata = new ExceptionMetadata(fetchedPartition, fetchedOffset, e);
```

Comment:
This try/catch is needed because when `PartitionRecords.fetchRecords` is thrown, it is possible that there has already been some other records fetched from previous `PartitionRecords` in this case, we need to return those fetched records and delay the exception.